packages/babel-plugin-transform-react-constant-elements/src/index.ts

```diff
@@ -61,54 +62,70 @@ export default declare((api, options) => {
         return stop();
       }
 
-      // Ignore identifiers & JSX expressions.
+      // Ignore JSX expressions and immutable values.
       if (
         path.isJSXIdentifier() ||
         path.isJSXMemberExpression() ||
-        path.isJSXNamespacedName()
+        path.isJSXNamespacedName() ||
+        path.isImmutable()
       ) {
         return;
       }
 
+      // Ignore constant bindings.
       if (path.isIdentifier()) {
         const binding = path.scope.getBinding(path.node.name);
         if (binding && binding.constant) return;
       }
 
-      if (!path.isImmutable()) {
-        // If it's not immutable, it may still be a pure expression, such as string concatenation.
```

Comment:
This was not actually changed; I suggest hiding whitespace changes when reviewing.